### ISSUE TITLE
Parsing batch item json fails

### ISSUE THREAD

**Issue by johavik** [MvdDonk/brewfather#18]
It seems like parsing of json data for specific batches fails.
I've turned on debug logging, the log is included below. Please let me know if there is any additional testing I can help with.

```
2024-12-08 09:54:59.135 DEBUG (MainThread) [custom_components.brewfather.coordinator] Updating data via library
2024-12-08 09:54:59.135 DEBUG (MainThread) [custom_components.brewfather.coordinator] Updating data...
2024-12-08 09:54:59.135 DEBUG (MainThread) [custom_components.brewfather.connection] Making api call to: https://api.brewfather.app/v2/batches/?status=Fermenting
2024-12-08 09:54:59.526 DEBUG (MainThread) [custom_components.brewfather.connection] Api call response: [{"_id":"tXyKX8BIL3jBVKT93g56UQ6VvWd9w1","batchNo":34,"brewDate":1733526000000,"brewer":"Jo H","name":"Batch","recipe":{"name":"Kauk cerveza sour edition"},"status":"Fermenting"}]
2024-12-08 09:54:59.526 DEBUG (MainThread) [custom_components.brewfather.connection] Making api call to: https://api.brewfather.app/v2/batches/tXyKX8BIL3jBVKT93g56UQ6VvWd9w1?include=recipe.fermentation,notes,measuredOg
2024-12-08 09:54:59.870 DEBUG (MainThread) [custom_components.brewfather.connection] Api call response: {"_id":"tXyKX8BIL3jBVKT93g56UQ6VvWd9w1","name":"Batch","batchNo":34,"status":"Fermenting","brewer":"Jo H","brewDate":1733526000000,"recipe":{"name":"Kauk cerveza sour edition","fermentation":{"_id":"ROH9v0qw1fSb4toWwCSAigYpCpIScc","name":"Sour + novalager","steps":[{"actualTime":1733526000000,"pressure":0,"stepTemp":25,"displayPressure":0,"stepTime":1,"name":"Sour","type":"Primary","displayStepTemp":25,"ramp":null},{"name":"novalager","pressure":0,"actualTime":1733612400000,"ramp":0,"type":"Primary","stepTemp":18,"displayPressure":0,"displayStepTemp":18,"stepTime":5},{"actualTime":1734217200000,"name":"Diacetyl","displayStepTemp":19,"stepTime":2,"pressure":10.152642,"type":"Primary","ramp":2,"displayPressure":0.7,"stepTemp":19,"actualRampTime":1734044400000},{"stepTemp":2,"actualRampTime":1734390000000,"pressure":17.404529,"stepTime":2,"actualTime":1734735600000,"name":"Cold crash","type":"Cold Crash","displayPressure":1.2,"displayStepTemp":2,"ramp":4}],"pressurized":true}},"notes":[{"timestamp":1733580762888,"note":"","type":"statusChanged","status":"Fermenting"},{"timestamp":1733566571144,"note":"","type":"statusChanged","status":"Brewing"}]}
2024-12-08 09:54:59.871 ERROR (MainThread) [custom_components.brewfather.connection] Unable read or parse json response: 
2024-12-08 09:54:59.871 DEBUG (MainThread) [custom_components.brewfather.coordinator] Finished fetching brewfather data in 0.736 seconds (success: True)
2024-12-08 09:54:59.871 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry Brewfather for brewfather
Traceback (most recent call last):
  File "/config/custom_components/brewfather/connection.py", line 91, in get_api_response
    return parseJson(jsonData)
  File "/config/custom_components/brewfather/models/batch_item.py", line 365, in batch_item_from_dict
    return BatchItem.from_dict(s)
           ~~~~~~~~~~~~~~~~~~~^^^
  File "/config/custom_components/brewfather/models/batch_item.py", line 293, in from_dict
    recipe = from_union([Recipe.from_dict, from_none], obj.get("recipe"))
  File "/config/custom_components/brewfather/models/batch_item.py", line 27, in from_union
    assert False
           ^^^^^
AssertionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 640, in __async_setup_with_context
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/brewfather/__init__.py", line 35, in async_setup_entry
    await coordinator.async_config_entry_first_refresh()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 306, in async_config_entry_first_refresh
    await self._async_refresh(
        log_failures=False, raise_on_auth_failed=True, raise_on_entry_error=True
    )
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 379, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ om
  File "/config/custom_components/brewfather/coordinator.py", line 80, in _async_update_data
    data = await self.update()
           ^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/brewfather/coordinator.py", line 91, in update
    batchData = await self.connection.get_batch(batch.id)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/brewfather/connection.py", line 62, in get_batch
    batch = await self.get_api_response(url, batch_item_from_dict)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/brewfather/connection.py", line 94, in get_api_response
    exit(1)
    ~~~~^^^
  File "<frozen _sitebuiltins>", line 26, in __call__
SystemExit: 1
```

**Comment by MvdDonk** [comment]
Hi @johavik thanks for reporting this issue!
I have created a fix in version 2.0.1. Please let me know if it works, thanks.

**Comment by johavik** [comment]
Works fine, thank you for the prompt fix :)
